x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/InferenceActionProxy.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.inference.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * This action is used when making a REST request to the inference API. The transport handler
+ * will then look at the task type in the params (or retrieve it from the persisted model if it wasn't
+ * included in the params) to determine where this request should be routed. If the task type is chat completion
+ * then it will be routed to the unified chat completion handler by creating the {@link UnifiedCompletionAction}.
+ * If not, it will be passed along to {@link InferenceAction}.
+ */
+public class InferenceActionProxy extends ActionType<InferenceAction.Response> {
+    public static final InferenceActionProxy INSTANCE = new InferenceActionProxy();
+    public static final String NAME = "cluster:monitor/xpack/inference";
```

Comment:
I'm not sure if it is safe to use the old  `InferenceAction` name can be used here when it has different request and response classes. I'm thinking about a mixed cluster. To be safe please give it a new name.

Get, Put and Delete are `cluster:monitor/xpack/inference/[get|put|delete]`  this could be `cluster:monitor/xpack/inference/post` or cluster:monitor/xpack/inference/inference`

Reply:
Ah ok, I'll switch to `post` 👍